scipy/sparse/linalg/_dsolve/_superlumodule.c

```diff
@@ -263,6 +264,170 @@ static PyObject *Py_gstrf(PyObject * self, PyObject * args,
+    (PyArrayObject*)L_nzvals, (PyArrayObject*)L_colind, (PyArrayObject*)L_rowptr, L_type, SLU_SC, SLU_TRLU, L_col_to_sup,L_sup_to_col);
+  if(L_conv_err){
+    return NULL;
+  }
+  int U_conv_err = SparseFormat_from_spMatrix(&U_super, U_N, U_N, U_nnz, 0,
+    (PyArrayObject*)U_nzvals, (PyArrayObject*)U_rowind, (PyArrayObject*)U_colptr, U_type, SLU_NC, SLU_TRU, NULL,NULL);
+  if(U_conv_err){
+    Destroy_SuperMatrix_Store((SuperMatrix*)&L_super);
+    return NULL;
+  }
+
+  /* Read right-hand-side/solution vector. */
+  PyArrayObject* X_arr = (PyArrayObject*)PyArray_FROMANY(
+      (PyObject*)X_py, L_type, 1, 2,
+      NPY_ARRAY_F_CONTIGUOUS | NPY_ARRAY_ENSURECOPY);
+  if (X_arr == NULL){
+      SUPERLU_FREE((void*)L_col_to_sup); SUPERLU_FREE((void*)L_sup_to_col);
+      Destroy_SuperMatrix_Store((SuperMatrix*)&L_super);
+      Destroy_SuperMatrix_Store((SuperMatrix*)&U_super);
+      return NULL;
+  }
+  if (PyArray_DIM((PyArrayObject*)X_arr, 0) != L_N) {
+      PyErr_SetString(PyExc_ValueError,
+                      "right hand side array has invalid shape");
+      SUPERLU_FREE((void*)L_col_to_sup); SUPERLU_FREE((void*)L_sup_to_col);
+      Destroy_SuperMatrix_Store((SuperMatrix*)&L_super);
+      Destroy_SuperMatrix_Store((SuperMatrix*)&U_super);
+      Py_DECREF(X_arr);
+      return NULL;
+  }
+
+  SuperMatrix X;
+  if (DenseSuper_from_Numeric((SuperMatrix*)&X, (PyObject*)X_arr)) {
+      SUPERLU_FREE((void*)L_col_to_sup); SUPERLU_FREE((void*)L_sup_to_col);
+      Destroy_SuperMatrix_Store((SuperMatrix*)&L_super);
+      Destroy_SuperMatrix_Store((SuperMatrix*)&U_super);
+      Py_DECREF(X_arr);
+      return NULL;
+  } /* X and X_arr share the same data but X_arr "owns" it. */
+
+  /* Call SuperLU functions. */
+  int info=0;
+  SuperLUStat_t stat = { 0 };
+  StatInit((SuperLUStat_t *)&stat);
+  int* perm_c = intMalloc(L_N);
+  for(int i=0; i<L_N; i++){
+    perm_c[i] = i;
+  }
+  int* perm_r = perm_c;
+
+  jmpbuf_ptr = (volatile jmp_buf *)superlu_python_jmpbuf();
+  SLU_BEGIN_THREADS;
+  if (setjmp(*(jmp_buf*)jmpbuf_ptr)) {
+      SLU_END_THREADS;
+      goto fail;
+  }
+  gstrs(L_type,
+        trans, &L_super, &U_super, perm_c, perm_r,
+        (SuperMatrix *)&X, (SuperLUStat_t *)&stat, (int *)&info);
+  SLU_END_THREADS;
+
+  if (info) {
+      PyErr_SetString(PyExc_SystemError,
+                      "gstrs was called with invalid arguments");
+      goto fail;
+  }
+
+  /* Deallocations and return. */
+  SUPERLU_FREE((void*)L_col_to_sup); SUPERLU_FREE((void*)L_sup_to_col);
+  SUPERLU_FREE((void*)perm_c);
+  Destroy_SuperMatrix_Store(&L_super);
+  Destroy_SuperMatrix_Store(&U_super);
+  XStatFree((SuperLUStat_t *)&stat);
+
+  return Py_BuildValue("Ni", X_arr, info);
+
+  fail:
+  SUPERLU_FREE((void*)L_col_to_sup); SUPERLU_FREE((void*)L_sup_to_col);
+  SUPERLU_FREE((void*)perm_c);
+  Destroy_SuperMatrix_Store(&L_super);
+  Destroy_SuperMatrix_Store(&U_super);
+  XStatFree((SuperLUStat_t *)&stat);
+  Py_DECREF(X_arr);
+  return NULL;
```

Comment:
```suggestion
     SUPERLU_FREE((void*)L_col_to_sup); SUPERLU_FREE((void*)L_sup_to_col);
     SUPERLU_FREE((void*)perm_c);
     Destroy_SuperMatrix_Store(&L_super);
     Destroy_SuperMatrix_Store(&U_super);
     XStatFree((SuperLUStat_t *)&stat);
     Py_DECREF(X_arr);
     return NULL;
```

Could you also check your C code updates for indentations consistent with the rest of the code? There is no general C style guideline in scipy, but at least in one file we should aim for consistency.

Also the workflows should get approved once more due to the recent test updates.

Reply:
done in 904788fb5cc88a704cdea4680550658030c04df4